In [2]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [3]:
from google.colab import files
uploaded = files.upload()


Saving car_crash_train.csv to car_crash_train.csv


In [4]:
df = pd.read_csv("car_crash_train.csv")
print(df)

      Crash Speed (km/h)  Impact Angle (degrees) Airbag Deployed  \
0                    182                      69             Yes   
1                    192                      73             Yes   
2                    114                      77              No   
3                     93                      14              No   
4                    170                      96              No   
...                  ...                     ...             ...   
3995                  36                      13             Yes   
3996                  29                      67             Yes   
3997                 156                      50             Yes   
3998                 143                      23              No   
3999                 128                     115             Yes   

     Seatbelt Used Weather Conditions Road Conditions   Crash Type  \
0               No              Clear             Dry     Rollover   
1               No              Clear      

In [5]:
# Basic info
df.info()

# Summary statistics for numeric features
df.describe()

# Summary for categorical features
df.describe(include=['object'])

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4000 entries, 0 to 3999
Data columns (total 19 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Crash Speed (km/h)         4000 non-null   int64  
 1   Impact Angle (degrees)     4000 non-null   int64  
 2   Airbag Deployed            4000 non-null   object 
 3   Seatbelt Used              4000 non-null   object 
 4   Weather Conditions         4000 non-null   object 
 5   Road Conditions            4000 non-null   object 
 6   Crash Type                 4000 non-null   object 
 7   Vehicle Type               4000 non-null   object 
 8   Vehicle Age (years)        4000 non-null   int64  
 9   Brake Condition            4000 non-null   object 
 10  Tire Condition             4000 non-null   object 
 11  Driver Age                 4000 non-null   int64  
 12  Driver Experience (years)  4000 non-null   int64  
 13  Alcohol Level (BAC%)       4000 non-null   float

,Airbag Deployed,Seatbelt Used,Weather Conditions,Road Conditions,Crash Type,Vehicle Type,Brake Condition,Tire Condition,Distraction Level,Time of Day,Traffic Density,Severity
count,4000,4000,4000,4000,4000,4000,4000,4000,1640,4000,4000,4000
unique,2,2,4,4,4,4,2,2,3,3,3,3
top,Yes,Yes,Clear,Dry,Side impact,Sedan,Good,Good,Phone,Afternoon,Medium,Minor Injury
freq,2817,3174,2345,2792,1413,1644,3369,3225,773,1579,1979,2756


In [6]:
import pandas as pd
from sklearn.preprocessing import LabelEncoder

# List of categorical features to encode (excluding target)
categorical_features = ['Airbag Deployed', 'Seatbelt Used', 'Weather Conditions',
                        'Road Conditions', 'Crash Type', 'Vehicle Type',
                        'Brake Condition', 'Tire Condition', 'Distraction Level',
                        'Time of Day', 'Traffic Density']

# One-Hot Encoding
df_encoded = pd.get_dummies(df, columns=categorical_features, drop_first=True)

# Encode target variable
le = LabelEncoder()
y = le.fit_transform(df_encoded['Severity'])

# Drop target from features
X = df_encoded.drop('Severity', axis=1)

print("Shape of features:", X.shape)
print("Classes in target:", le.classes_)

Shape of features: (4000, 29)
Classes in target: ['Fatal' 'Minor Injury' 'Severe Injury']


In [7]:
from sklearn.preprocessing import StandardScaler

numeric_features = ['Crash Speed (km/h)', 'Impact Angle (degrees)',
                    'Vehicle Age (years)', 'Driver Age',
                    'Driver Experience (years)', 'Alcohol Level (BAC%)',
                    'Visibility Distance (m)']

scaler = StandardScaler()
X[numeric_features] = scaler.fit_transform(X[numeric_features])

In [8]:
#knn
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix

# 1. Split the training dataset into 80% train / 20% validation
X_train_split, X_val, y_train_split, y_val = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)

# 2. Initialize KNN (you can adjust n_neighbors later)
knn = KNeighborsClassifier(n_neighbors=5)

# 3. Train KNN on 80% training split
knn.fit(X_train_split, y_train_split)

# 4. Predict on validation set (20%)
y_pred = knn.predict(X_val)

# 5. Evaluate performance
accuracy = accuracy_score(y_val, y_pred)
print("KNN Accuracy:", accuracy)
print("\nClassification Report:\n", classification_report(y_val, y_pred, target_names=le.classes_))
print("\nConfusion Matrix:\n", confusion_matrix(y_val, y_pred))

KNN Accuracy: 0.63375

Classification Report:
                precision    recall  f1-score   support

        Fatal       0.07      0.02      0.04        41
 Minor Injury       0.69      0.88      0.77       551
Severe Injury       0.26      0.10      0.14       208

     accuracy                           0.63       800
    macro avg       0.34      0.33      0.32       800
 weighted avg       0.54      0.63      0.57       800


Confusion Matrix:
 [[  1  38   2]
 [  9 486  56]
 [  4 184  20]]


KNN achieves moderate overall accuracy (63.4%). Minor Injury, the majority class, is predicted very well with recall of 0.88 and F1-score of 0.77. Fatal and Severe crashes are detected very poorly, with recall less than 0.10 for both classes. The confusion matrix shows the model over-predicts Minor Injury, misclassifying almost all Fatal and Severe crashes.

In [10]:
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix

# 1. Split the training dataset into 80% train / 20% validation
X_train_split, X_val, y_train_split, y_val = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)

# 2. Initialize KNN (you can adjust n_neighbors later)
knn = KNeighborsClassifier(n_neighbors=10)

# 3. Train KNN on 80% training split
knn.fit(X_train_split, y_train_split)

# 4. Predict on validation set (20%)
y_pred = knn.predict(X_val)

# 5. Evaluate performance
accuracy = accuracy_score(y_val, y_pred)
print("KNN Accuracy:", accuracy)
print("\nClassification Report:\n", classification_report(y_val, y_pred, target_names=le.classes_))
print("\nConfusion Matrix:\n", confusion_matrix(y_val, y_pred))

KNN Accuracy: 0.6725

Classification Report:
                precision    recall  f1-score   support

        Fatal       0.00      0.00      0.00        41
 Minor Injury       0.69      0.96      0.80       551
Severe Injury       0.26      0.04      0.07       208

     accuracy                           0.67       800
    macro avg       0.32      0.33      0.29       800
 weighted avg       0.54      0.67      0.57       800


Confusion Matrix:
 [[  0  39   2]
 [  0 530  21]
 [  0 200   8]]


/usr/local/lib/python3.12/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))
/usr/local/lib/python3.12/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))
/usr/local/lib/python3.12/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))


In [12]:
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix

# 1. Split the training dataset into 80% train / 20% validation
X_train_split, X_val, y_train_split, y_val = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)

# 2. Initialize KNN (you can adjust n_neighbors later)
knn = KNeighborsClassifier(n_neighbors=15)

# 3. Train KNN on 80% training split
knn.fit(X_train_split, y_train_split)

# 4. Predict on validation set (20%)
y_pred = knn.predict(X_val)

# 5. Evaluate performance
accuracy = accuracy_score(y_val, y_pred)
print("KNN Accuracy:", accuracy)
print("\nClassification Report:\n", classification_report(y_val, y_pred, target_names=le.classes_))
print("\nConfusion Matrix:\n", confusion_matrix(y_val, y_pred))

KNN Accuracy: 0.68

Classification Report:
                precision    recall  f1-score   support

        Fatal       0.00      0.00      0.00        41
 Minor Injury       0.69      0.98      0.81       551
Severe Injury       0.21      0.01      0.03       208

     accuracy                           0.68       800
    macro avg       0.30      0.33      0.28       800
 weighted avg       0.53      0.68      0.56       800


Confusion Matrix:
 [[  0  40   1]
 [  0 541  10]
 [  0 205   3]]


/usr/local/lib/python3.12/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))
/usr/local/lib/python3.12/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))
/usr/local/lib/python3.12/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))


In [12]:
from sklearn.model_selection import train_test_split
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix

# 1. Split the training dataset into 80% train / 20% validation
X_train_split, X_val, y_train_split, y_val = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)

# 2. Initialize KNN (you can adjust n_neighbors later)
knn = KNeighborsClassifier(n_neighbors=20)

# 3. Train KNN on 80% training split
knn.fit(X_train_split, y_train_split)

# 4. Predict on validation set (20%)
y_pred = knn.predict(X_val)

# 5. Evaluate performance
accuracy = accuracy_score(y_val, y_pred)
print("KNN Accuracy:", accuracy)
print("\nClassification Report:\n", classification_report(y_val, y_pred, target_names=le.classes_))
print("\nConfusion Matrix:\n", confusion_matrix(y_val, y_pred))

KNN Accuracy: 0.68875

Classification Report:
                precision    recall  f1-score   support

        Fatal       0.00      0.00      0.00        41
 Minor Injury       0.69      0.99      0.81       551
Severe Injury       0.50      0.02      0.04       208

     accuracy                           0.69       800
    macro avg       0.40      0.34      0.28       800
 weighted avg       0.61      0.69      0.57       800


Confusion Matrix:
 [[  0  41   0]
 [  0 547   4]
 [  0 204   4]]


/usr/local/lib/python3.12/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))
/usr/local/lib/python3.12/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))
/usr/local/lib/python3.12/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))


In [15]:
# -------------------------------
# Decision Tree Training & Validation
# -------------------------------

# 1. Imports (only necessary ones for Decision Tree)
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix

# 2. Split the preprocessed training dataset into 80% training / 20% validation
X_train_split, X_val, y_train_split, y_val = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)

print("Training samples:", X_train_split.shape[0])
print("Validation samples:", X_val.shape[0])

# 3. Initialize the Decision Tree
dtree = DecisionTreeClassifier(random_state=42)

# 4. Train the Decision Tree on the training split
dtree.fit(X_train_split, y_train_split)

# 5. Predict on the validation set
y_pred_dt = dtree.predict(X_val)

# 6. Evaluate model performance
accuracy_dt = accuracy_score(y_val, y_pred_dt)
print("\nDecision Tree Accuracy:", accuracy_dt)

print("\nClassification Report:\n", classification_report(
    y_val, y_pred_dt, target_names=le.classes_
))

cm_dt = confusion_matrix(y_val, y_pred_dt)
print("\nConfusion Matrix:\n", cm_dt)

Training samples: 3200
Validation samples: 800

Decision Tree Accuracy: 0.52375

Classification Report:
                precision    recall  f1-score   support

        Fatal       0.05      0.07      0.06        41
 Minor Injury       0.68      0.66      0.67       551
Severe Injury       0.26      0.26      0.26       208

     accuracy                           0.52       800
    macro avg       0.33      0.33      0.33       800
 weighted avg       0.54      0.52      0.53       800


Confusion Matrix:
 [[  3  30   8]
 [ 41 362 148]
 [ 12 142  54]]


The Decision Tree captures the majority class reasonably well but fails to detect minority class crashes, which is critical in safety applications.

In [18]:
# -------------------------------
# Random Forest Classifier Training & Validation
# -------------------------------

from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix
import pandas as pd

# 1. Split the preprocessed training dataset into 80% train / 20% validation
X_train_split, X_val, y_train_split, y_val = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)

print("Training samples:", X_train_split.shape[0])
print("Validation samples:", X_val.shape[0])

# 2. Initialize and train Random Forest
rf = RandomForestClassifier(n_estimators=100, random_state=42)
rf.fit(X_train_split, y_train_split)

# 3. Predict and evaluate on the validation set
y_pred_rf = rf.predict(X_val)

accuracy_rf = accuracy_score(y_val, y_pred_rf)
print("\nRandom Forest Accuracy:", accuracy_rf)

print("\nClassification Report:\n", classification_report(
    y_val, y_pred_rf, target_names=le.classes_
))

cm_rf = confusion_matrix(y_val, y_pred_rf)
print("\nConfusion Matrix:\n", cm_rf)

Training samples: 3200
Validation samples: 800

Random Forest Accuracy: 0.6875

Classification Report:
                precision    recall  f1-score   support

        Fatal       0.00      0.00      0.00        41
 Minor Injury       0.69      1.00      0.81       551
Severe Injury       0.00      0.00      0.00       208

     accuracy                           0.69       800
    macro avg       0.23      0.33      0.27       800
 weighted avg       0.47      0.69      0.56       800


Confusion Matrix:
 [[  0  41   0]
 [  0 550   1]
 [  0 208   0]]


/usr/local/lib/python3.12/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))
/usr/local/lib/python3.12/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))
/usr/local/lib/python3.12/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))


In [19]:
# -------------------------------
# Logistic Regression Training & Validation
# -------------------------------

from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix

# 1. Split the preprocessed training dataset into 80% train / 20% validation
X_train_split, X_val, y_train_split, y_val = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)

print("Training samples:", X_train_split.shape[0])
print("Validation samples:", X_val.shape[0])

# 2. Initialize and train Logistic Regression
logreg = LogisticRegression(max_iter=1000, random_state=42, multi_class='multinomial')
logreg.fit(X_train_split, y_train_split)

# 3. Predict and evaluate on the validation set
y_pred_lr = logreg.predict(X_val)

accuracy_lr = accuracy_score(y_val, y_pred_lr)
print("\nLogistic Regression Accuracy:", accuracy_lr)

print("\nClassification Report:\n", classification_report(
    y_val, y_pred_lr, target_names=le.classes_
))

cm_lr = confusion_matrix(y_val, y_pred_lr)
print("\nConfusion Matrix:\n", cm_lr)

Training samples: 3200
Validation samples: 800


/usr/local/lib/python3.12/dist-packages/sklearn/linear_model/_logistic.py:1247: FutureWarning: 'multi_class' was deprecated in version 1.5 and will be removed in 1.7. From then on, it will always use 'multinomial'. Leave it to its default value to avoid this warning.
  warnings.warn(



Logistic Regression Accuracy: 0.68875

Classification Report:
                precision    recall  f1-score   support

        Fatal       0.00      0.00      0.00        41
 Minor Injury       0.69      1.00      0.82       551
Severe Injury       0.00      0.00      0.00       208

     accuracy                           0.69       800
    macro avg       0.23      0.33      0.27       800
 weighted avg       0.47      0.69      0.56       800


Confusion Matrix:
 [[  0  41   0]
 [  0 551   0]
 [  0 208   0]]


/usr/local/lib/python3.12/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))
/usr/local/lib/python3.12/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))
/usr/local/lib/python3.12/dist-packages/sklearn/metrics/_classification.py:1565: UndefinedMetricWarning: Precision is ill-defined and being set to 0.0 in labels with no predicted samples. Use `zero_division` parameter to control this behavior.
  _warn_prf(average, modifier, f"{metric.capitalize()} is", len(result))


In [20]:
# -------------------------------
# Naïve Bayes Training & Validation
# -------------------------------

from sklearn.naive_bayes import GaussianNB
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix

# 1. Split the preprocessed training dataset into 80% train / 20% validation
X_train_split, X_val, y_train_split, y_val = train_test_split(
    X, y, test_size=0.2, random_state=42, stratify=y
)

print("Training samples:", X_train_split.shape[0])
print("Validation samples:", X_val.shape[0])

# 2. Initialize and train Naïve Bayes
nb = GaussianNB()
nb.fit(X_train_split, y_train_split)

# 3. Predict and evaluate on the validation set
y_pred_nb = nb.predict(X_val)

accuracy_nb = accuracy_score(y_val, y_pred_nb)
print("\nNaïve Bayes Accuracy:", accuracy_nb)

print("\nClassification Report:\n", classification_report(
    y_val, y_pred_nb, target_names=le.classes_, zero_division=0
))

cm_nb = confusion_matrix(y_val, y_pred_nb)
print("\nConfusion Matrix:\n", cm_nb)

Training samples: 3200
Validation samples: 800

Naïve Bayes Accuracy: 0.655

Classification Report:
                precision    recall  f1-score   support

        Fatal       0.00      0.00      0.00        41
 Minor Injury       0.69      0.93      0.79       551
Severe Injury       0.31      0.06      0.10       208

     accuracy                           0.66       800
    macro avg       0.33      0.33      0.30       800
 weighted avg       0.55      0.66      0.57       800


Confusion Matrix:
 [[  0  38   3]
 [ 14 511  26]
 [  1 194  13]]


In [37]:
# -------------------------------
# Final Test Set Predictions for All 5 Models
# -------------------------------

import pandas as pd
from sklearn.metrics import accuracy_score, classification_report, confusion_matrix

# 1. Load test CSV
test_df = pd.read_csv("car_crash_test.csv")

# 2. Separate features and true labels
X_test = test_df.drop('Severity', axis=1)
y_test = test_df['Severity']

# 3. One-hot encode categorical features (same as training)
categorical_features = ['Airbag Deployed', 'Seatbelt Used', 'Weather Conditions',
                        'Road Conditions', 'Crash Type', 'Vehicle Type',
                        'Brake Condition', 'Tire Condition', 'Distraction Level',
                        'Time of Day', 'Traffic Density']

X_test_encoded = pd.get_dummies(X_test, columns=categorical_features, drop_first=True)

# Ensure test set has same columns as training set
missing_cols = set(X.columns) - set(X_test_encoded.columns)
for col in missing_cols:
    X_test_encoded[col] = 0

# Reorder columns to match training features
X_test_encoded = X_test_encoded[X.columns]

# 4. Scale numeric features using the same scaler as training
numeric_features = ['Crash Speed (km/h)', 'Impact Angle (degrees)', 'Vehicle Age (years)',
                    'Driver Age', 'Driver Experience (years)', 'Alcohol Level (BAC%)',
                    'Visibility Distance (m)']
X_test_encoded[numeric_features] = scaler.transform(X_test_encoded[numeric_features])

# 5. Encode true test labels to numeric
y_test_encoded = le.transform(y_test)

# 6. Dictionary of trained models
models = {
    "KNN": knn,
    "Decision Tree": dtree,
    "Random Forest": rf,
    "Logistic Regression": logreg,
    "Naive Bayes": nb
}

# 7. Predict and evaluate
for name, model in models.items():
    y_pred = model.predict(X_test_encoded)

    print("--------------------------------------------------")
    print(f"{name} on Test Set:")

    # Accuracy
    acc = accuracy_score(y_test_encoded, y_pred)
    print("Accuracy:", acc)

    # Classification report
    print("\nClassification Report:\n", classification_report(
        y_test_encoded, y_pred, target_names=le.classes_, zero_division=0
    ))

    # Confusion matrix
    cm = confusion_matrix(y_test_encoded, y_pred)
    print("Confusion Matrix:\n", cm)

# 8. Optional: Prepare submission CSV (if needed)
# Convert numeric predictions of one model to original labels for submission
y_pred_labels = le.inverse_transform(dtree.predict(X_test_encoded))
submission = test_df.copy()
submission['Predicted_Severity'] = y_pred_labels
submission_final = submission[['Predicted_Severity']]  # adjust columns as per sample submission
submission_final.to_csv("decision_tree_test_predictions.csv", index=False)

--------------------------------------------------
KNN on Test Set:
Accuracy: 0.706

Classification Report:
                precision    recall  f1-score   support

        Fatal       0.00      0.00      0.00        41
 Minor Injury       0.71      0.99      0.83       710
Severe Injury       0.30      0.01      0.02       249

     accuracy                           0.71      1000
    macro avg       0.34      0.33      0.28      1000
 weighted avg       0.58      0.71      0.59      1000

Confusion Matrix:
 [[  0  41   0]
 [  0 703   7]
 [  0 246   3]]
--------------------------------------------------
Decision Tree on Test Set:
Accuracy: 0.528

Classification Report:
                precision    recall  f1-score   support

        Fatal       0.05      0.10      0.07        41
 Minor Injury       0.71      0.63      0.67       710
Severe Injury       0.26      0.31      0.29       249

     accuracy                           0.53      1000
    macro avg       0.34      0.35      0.